documentation/faq.asciidoc

```diff
@@ -384,4 +384,29 @@ This may be the case when Debezium (or rather the  MongoDB change stream cursor)
 Note that depending on the used `capture.mode` this issue can still manifest even when the actual value of the *stored* documents is significantly lower. 
 
 To mitigate this issue, refer to the documentation of link:/documentation/reference/stable/connectors/mongodb.html#mongodb-property-cursor-oversize-handling-mode[cursor.oversize.handling.mode] and
-link:/documentation/reference/stable/connectors/mongodb.html#mongodb-property-cursor-oversize-skip-threshold[cursor.oversize.skip.threshold] properties.
\ No newline at end of file
+link:/documentation/reference/stable/connectors/mongodb.html#mongodb-property-cursor-oversize-skip-threshold[cursor.oversize.skip.threshold] properties.
+
+== Debezium engine applications fails with "Unable to find minimal snapshot lock mode. Please check your configuration" error
+
+Debezium uses the SPI via the ServiceLoader to load different implementations depending on the connector type or for custom implementations.
+
+When an interface has multiple implementations — for instance, `io.debezium.snapshot.spi.SnapshotLock`, which has a default implementation in the core and specific ones for each connector — you must explicitly configure your build tools to merge the `META-INF/services files`.
+If this step is missed, Debezium may fail to locate the required implementation.
+
+To address this, instruct your build tools accordingly. For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
+you need to add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer:
+
+[source,xml]
+----
+...
+<configuration>
+ <transformers>
+    ...
+    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+    ...
+ </transformers>
+...
+</configuration>
+----
+
+Alternatively, if using the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in] you can utilize the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```

Comment:
```suggestion
Debezium uses uses the SPI via the ServiceLoader to load the implementation.
The implementation can be based on the connector type, or it can be a custom implementation.

Some interfaces have multiple implementations.
For example, `io.debezium.snapshot.spi.SnapshotLock` has a default implementation in the core, and specific implementations for each connector.
To ensure that {prodname} can locate the required implementation, you must explicitly configure your build tools to merge the `META-INF/services files`. 


For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer, as shown in the following example:

[source,xml]
----
...
<configuration>
 <transformers>
    ...
    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
    ...
 </transformers>
...
</configuration>
----

Alternatively, if you use the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in], you can use the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```